lib/web_catalog/api_extractor.es6.js

```diff
@@ -0,0 +1,406 @@
+          if (prototypeId) {
+            prototypeProps = this.getObjectProperties_(og, prototypeId);
+          }
+          this.arrayMerge_(ownProperties, prototypeProps);
+          let meaningfulProps = this.setMinus_(ownProperties,
+            this.buildInFunctionProperties);
+          if (meaningfulProps.length > 0) {
+            // If this function object contains meaningful properties.
+            if (!map[interfaceName]) map[interfaceName] = [];
+            this.arrayMerge_(map[interfaceName],
+              this.getObjectProperties_(og, id),
+              this.getObjectProperties_(og, prototypeId));
+          }
+          if (og.getPrototype(id)) {
+            this.extractAPI_(og, map, og.getPrototype(id),
+              `${name}.__proto__`, visited, {proto: true});
+          }
+        } else if (!og.isFunction(id)) {
+          // Object is not a first level object or prototype function,
+          // nor is it a function.
+          let ownProperty = this.getObjectProperties_(og, id);
+          if (this.setMinus_(ownProperty,
+            this.buildInObjectProperties).length > 0) {
+            let className = this.getClassName_(id, og);
+            if ((!className || className === 'Object') && firstLevel) {
+              // If object is at first level and its class name is "Object",
+              // this object is considered as an object library,
+              // such as MATH, console.
+              let prototypeId = og.lookup('prototype', id);
+              // In Firefox, Edge, (maybe some other vendors or versions),
+              // objects can have prototype properties.
+              let prototypeProps = [];
+              if (prototypeId) {
+                prototypeProps = this.getObjectProperties_(og, prototypeId);
+              }
+              if (!map[name]) map[name] = [];
+              this.arrayMerge_(map[name],
+                this.getObjectProperties_(og, id), prototypeProps);
+            } else {
+              // This is not a object library, try to add its property to
+              // it's prototype. For example, document.body.style contains
+              // meaningful properties of CSSStyleDeclaration, then its
+              // properties will be added to CSSStyleDeclaration interface.
+              if (!className || className === 'Object') return;
+              // Skip if its class is Object, or cannot find it's class name.
+              // We don't want to add extra properties for Object.
+              if (!map[className]) map[className] = [];
+              this.arrayMerge_(map[className],
+                this.getObjectProperties_(og, id));
+            }
+          }
+          if (og.getPrototype(id)) {
+            // Visit object __proto__ if its __proto__ exists.
+            this.extractAPI_(og, map, og.getPrototype(id),
+              `${name}.__proto__`, visited, {proto: true});
+          }
+        }
+
+        // Don't visit __proto__'s properties
```

Comment:
"__proto__ is the object that is used in the lookup chain to resolve methods". I think the __proot__ object does not contain meaningful interface/instances in their properties. And we already add __proto__'s property to its instance, it will be not helpful to visit __proto__'s properties.